assembly/src/assembler/instruction/mod.rs

```diff
@@ -54,332 +54,382 @@ impl Assembler {
     fn compile_instruction_impl(
         &self,
         instruction: &Span<Instruction>,
-        span_builder: &mut BasicBlockBuilder,
+        basic_block_builder: &mut BasicBlockBuilder,
```

Comment:
As mentioned above, we should leave it as-is in this PR since we're consistent in our use of "basic block". IMO we should only switch to `block_builder` when we change all uses of "basic block" to "block".